src/vm/gccover.cpp

```diff
@@ -1324,6 +1324,11 @@ BOOL OnGcCoverageInterrupt(PCONTEXT regs)
     if (!codeInfo.IsValid())
         return(FALSE);
 
+#ifdef _TARGET_ARM_
+    _ASSERTE(((TADDR)instrPtr) & THUMB_CODE);
```

Comment:
You can do this without  ifdefs (and fewer casts):

```
PCODE pControlPc = GetIP(regs);
...
TADDR instrPtr = PCODEToPINSTR(pControlPC);
...
EECodeInfo codeInfo(pControlPc);
```

Reply:
I've changed that and also the place in DoGcStress where I have originally copied the change from.